src/main/java/edu/iis/powp/adapter/IPlotterAndDrawPanelControllerAdapter.java

```diff
@@ -27,16 +28,19 @@ public void setPosition(int x, int y)
     @Override
     public void drawTo(int x, int y)
     {
-        ILine line = LineFactory.getBasicLine();
-    	line.setStartCoordinates(this.startX, this.startY);
-        line.setEndCoordinates(x, y);
-
-		drawLine(line);
-    }
+		ILine line = LineFactory.getBasicLine();
+		line.setStartCoordinates(this.startX, this.startY);
+		line.setEndCoordinates(x, y);
+		
+		this.startX = x;
+        this.startY = y;
+        
+        ApplicationWithDrawer.getDrawPanelController().drawLine(line);
```

Comment:
1. Constructor injection should be used (using a parameter). This is unnecessary direct coupling with ApplicationDrawer class.
2. Inheritance is not used (no super drawLine calls). Should be removed!